core/mutator/mutator.go

```diff
@@ -55,7 +55,9 @@ type Mutator interface {
 // Mutation reads and writes mutations to the database.
 type Mutation interface {
 	// Read reads all mutations for a specific given mapID, epoch, and index.
-	Read(ctx context.Context, txn transaction.Txn, epoch int64, index []byte) ([]byte, error)
-	// Write saves the mutation in the database.
-	Write(ctx context.Context, txn transaction.Txn, epoch int64, index, mutation []byte) error
+	Read(ctx context.Context, txn transaction.Txn, epoch int64, index []byte) (uint64, []byte, error)
+	// Write saves the mutation in the database. Sequence might not be used
+	// by all databases engines. For instance, in MySQL and SQLite, sequence
+	// numbers are automatically added at the database level.
+	Write(ctx context.Context, txn transaction.Txn, sequence uint64, epoch int64, index, mutation []byte) error
```

Comment:
If possible, it would also be nice to avoid needing to know the current epoch when writing mutation here too.